super_gui/src/components/DoorManUI.tsx

```diff
@@ -43,12 +42,50 @@ const DoorManUI = () => {
     fetchBuildingLockers();
   }, []);
 
-  // Mock function to assign package
-  const assignPackage = () => {
-    if (!apartmentNumber || !selectedLocker) {
-      return;
+  // PATCH To Assign Packages
+  const assignPackage = async (lockerId: number, apartmentNumber: string) => {
+    try {
+      const res = await fetch(`http://localhost:8080/locker/${lockerId}`, {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          apartmentNumber: apartmentNumber,
+        }),
+      });
+      if (res.ok) {
+        const data = await res.json();
+
+        // Update the lockers status UI
+        setLockers((prevLockers) =>
+          prevLockers.map((locker) =>
```

Comment:
I think you get the updated locker object back with the request. IF so, you could use filter, instead of map. It would be something like this. Readabitlity is maybe better. Do not change it when you in a hurry.

`const lockersWithoutChangedLocker: PackageLocker = lockers.filter(locker => locker.id !== lockerId );
setLockers([ data.locker , ...lockersWithoutChangedLocker]);`